corehq/apps/app_manager/suite_xml/post_process/instances.py

```diff
@@ -283,6 +283,9 @@ def get_instance_factory(instance_name):
         scheme, _ = instance_name.split(':', 1)
     except ValueError:
         scheme = instance_name
+        # hack for selected cases
+        if "selected_cases" in instance_name:
```

Comment:
this change that was blocking for QA on staging is causing test failures-- from what I can see in multi-select case lists. @snopoke tagging you for your thoughts if it should be _all_ that contain `selected_cases`. Also @nospame for that QA bug, was it determined that it was a rouge QA app configuration and in that case can it be changed back to a less broad check?

Reply:
Yeah, it seems like the issue was a result of the config described [here](https://dimagi-dev.atlassian.net/browse/QA-5037?focusedCommentId=262219), which is probably not valid. I just checked by changing the "parent child selection" of the offending module and found the unrecognized `parent_selected_cases_show_multi` instance was no longer generated.

Reply:
Thanks for linking the config-- yeah that instance would have been fixed by this change

Reply:
@stephherbers I think this is fine as it is and is more robust the the current code so I'd like to keep it. But there are two other changes that are needed:

1. Remove 'selected_cases' and 'search_selected_cases' from the `INSTANCE_KWARGS_BY_ID` map
2. Update the `selected_cases_instances` method to add the instance name after the ref:

```python
@register_factory('selected_cases')
def selected_cases_instances(app, instance_name):
    return Instance(id=instance_name, src=f'jr://instance/selected-entities/{instance_name}')
```

Reply:
cool thank you Simon: https://github.com/dimagi/commcare-hq/pull/32554/commits/e9bd5f9a28388e9f675febc8fe076891208d9430 and https://github.com/dimagi/commcare-hq/pull/32554/commits/1d001ad6a021760e2cbcbe0dc143353cf9363ad9